source/core/parameters.cc

```diff
@@ -1331,17 +1343,17 @@ namespace Parameters
             if (op == "constant")
               {
                 surface_tension_model = SurfaceTensionModel::constant;
-                surface_tension_parameters.parse_parameters(prm);
+                surface_tension_parameters.parse_parameters(prm, dimensions);
               }
             else if (op == "linear")
               {
                 surface_tension_model = SurfaceTensionModel::linear;
-                surface_tension_parameters.parse_parameters(prm);
+                surface_tension_parameters.parse_parameters(prm, dimensions);
               }
             else if (op == "phase change")
               {
                 surface_tension_model = SurfaceTensionModel::phase_change;
-                surface_tension_parameters.parse_parameters(prm);
+                surface_tension_parameters.parse_parameters(prm, dimensions);
```

Comment:
Could we just call the parse_parameters once after the if/else ifs/else? Same for CH